projects/jdwp/defs/schema.py

```diff
@@ -32,30 +31,63 @@ class PrimitiveType(Enum):
     LOCATION = "location"
 
 
-class Array(Enum):
+class IntegralType(Enum):
+    """Integral type class."""
+
+    INT = "int"
+    BYTE = "byte"
+
+
+@dataclass(frozen=True)
+class ArrayLength:
+    """Array length class."""
+
+    type: IntegralType
+
+
+@dataclass(frozen=True)
+class Array:
     """Array class type."""
 
-    pass
+    element_type: Struct | TaggedUnion
+    length: Field[ArrayLength]
 
 
-class TaggedUnion(Enum):
+@dataclass(frozen=True)
+class TaggedUnion:
     """Tagged Union class type"""
 
-    pass
+    tag: Field[UnionTag]
+    cases: Dict[ModifierKind, Struct]
 
 
-Type = PrimitiveType | Array | TaggedUnion
+Type = (
+    PrimitiveType | Array | TaggedUnion | IntegralType | ArrayLength | Struct | UnionTag
+)
+
+T = TypeVar("T", bound=Type)
 
 
 @dataclass(frozen=True)
-class Field:
+class Field(Generic[T]):
     """Field class."""
 
     name: str
-    type: Type
+    type: T
     description: str
 
 
+M = TypeVar("M", bound=ModifierKind)
```

Comment:
To make this generic we want this to be a `TypeVar` that can be an arbitrary enum, not just `ModifierKind`. Would the following work ?

```
EnumT = TypeVar("M", bound=enum.Enum)
```